dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/programIndicator/DefaultProgramIndicatorSubqueryBuilder.java

```diff
@@ -0,0 +1,177 @@
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.hisp.dhis.analytics.event.data.programIndicator;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Date;
+
+import org.hisp.dhis.analytics.AggregationType;
+import org.hisp.dhis.analytics.event.ProgramIndicatorSubqueryBuilder;
+import org.hisp.dhis.commons.util.TextUtils;
+import org.hisp.dhis.program.AnalyticsType;
+import org.hisp.dhis.program.ProgramIndicator;
+import org.hisp.dhis.program.ProgramIndicatorService;
+import org.hisp.dhis.relationship.RelationshipType;
+import org.springframework.stereotype.Component;
+
+import com.google.common.base.Strings;
+
+@Component
+public class DefaultProgramIndicatorSubqueryBuilder
+    implements
+    ProgramIndicatorSubqueryBuilder
+{
+    private final static String ANALYTICS_TABLE_NAME = "analytics";
+
+    private final static String SUBQUERY_TABLE_ALIAS = "subax";
+
+    private final ProgramIndicatorService programIndicatorService;
+
+    public DefaultProgramIndicatorSubqueryBuilder( ProgramIndicatorService programIndicatorService )
+    {
+        checkNotNull( programIndicatorService );
+
+        this.programIndicatorService = programIndicatorService;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getAggregateClauseForProgramIndicator( ProgramIndicator pi, AnalyticsType outerSqlEntity,
+        Date earliestStartDate, Date latestDate )
+    {
+        return _getAggregateClauseForProgramIndicator( pi, null, outerSqlEntity, earliestStartDate, latestDate );
```

Comment:
Is the underscore a new convention I am not familiar with, or should the method be given another name?

Reply:
Ops, good catch, I was supposed to rename it.